src/main/java/movies/Server.java

```diff
@@ -109,23 +111,26 @@ private static Map<CrewRole, Long> crewCountForMovie(List<Credit> credits) {
 	}
 
 	private static Object moviesEndpoint(Request req, Response res) {
-		var movies = MOVIES.get().stream();
+		var movies = MOVIES.get();
 		movies = sortByDescReleaseDate(movies);
 		var query = req.queryParamOrDefault("q", req.queryParams("query"));
 		if (query != null) {
-			movies = movies.filter(m -> Pattern.matches(".*" + query.toUpperCase() + ".*", m.title.toUpperCase()));
+			movies = movies.stream().filter(m -> Pattern.matches(".*" + query.toUpperCase() + ".*", m.title.toUpperCase())).toList();
 		}
 		return replyJSON(res, movies);
 	}
 
-	private static Stream<Movie> sortByDescReleaseDate(Stream<Movie> movies) {
-		return movies.sorted(Comparator.comparing((Movie m) -> {
+	private static List<Movie> sortByDescReleaseDate(List<Movie> movies) {
+		var sortedMovies = new ArrayList<Movie>(movies);
+		Collections.sort(sortedMovies, Comparator.comparing((Movie m) -> {
 			try {
 				return LocalDate.parse(m.releaseDate);
 			} catch (Exception e) {
 				return LocalDate.MIN;
 			}
-		}).reversed());
+		}));
+		Collections.reverse(sortedMovies);
```

Comment:
Could we still use the reversed comparator approach ([link](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Comparator.html#reversed()))? It would save the extra additional in-place reverse (let's not add too many code inefficiencies lol).

Reply:
yep!